index.html

```diff
@@ -1131,203 +1128,64 @@ <h4 class="modal-title messagetitle">Changelog</h4>
-	$(".tech_industrialization").click(function() {research("industrialization")});
-	$(".tech_academicpublishing").click(function() {research("academicpublishing")});
-	$(".tech_patents").click(function() {research("patents")});
-	$(".tech_triforce").click(function() {research("triforce")});
-	$(".tech_logistics").click(function() {research("logistics")});
-	$(".tech_mineralcoal").click(function() {research("mineralcoal")});
-	$(".tech_electricity").click(function() {research("electricity")});
-	$(".tech_pyroprocessing").click(function() {research("pyroprocessing")});
-	$(".tech_education").click(function() {research("education")});
-	$(".tech_cementhydration").click(function() {research("cementhydration")});
-	$(".tech_workforce").click(function() {research("workforce")});
-	$(".tech_wargames").click(function() {research("wargames")});
-	$(".tech_militarization").click(function() {research("militarization")});
-	$(".tech_industrialrevolution").click(function() {research("industrialrevolution")});
-	$(".tech_luck").click(function() {research("luck")});
-
-	$(".craft_pickaxe").click(function() {crafting("pickaxe")});
-	$(".craft_spear").click(function() {crafting("spear")});
-	$(".craft_sword").click(function() {crafting("sword")});
-	$(".craft_block").click(function() {crafting("block")});
-	$(".craft_coin").click(function() {crafting("coin")});
-	$(".craft_bronze").click(function() {crafting("bronze")});
-	$(".craft_structure").click(function() {crafting("structure")});
-	$(".craft_armor").click(function() {crafting("armor")});
-	$(".craft_supplies").click(function() {crafting("supplies")});
-	$(".craft_chest").click(function() {crafting("chest")});
-	$(".craft_glass").click(function() {crafting("glass")});
-	$(".craft_bottle").click(function() {crafting("bottle")});
-	$(".craft_greatsword").click(function() {crafting("greatsword")});
-	$(".craft_frame").click(function() {crafting("frame")});
-	$(".craft_brick").click(function() {crafting("brick")});
-	$(".craft_gunpowder").click(function() {crafting("gunpowder")});
-	$(".craft_ammo").click(function() {crafting("ammo")});
-	$(".craft_musket").click(function() {crafting("musket")});
-	$(".craft_plate").click(function() {crafting("plate")});
-	$(".craft_engine").click(function() {crafting("engine")});
-	$(".craft_patent").click(function() {crafting("patent")});
-	$(".craft_book").click(function() {crafting("book")});
-	$(".craft_strategy").click(function() {crafting("strategy")});
+	
+	// This script replaces the long list of different build() calls below. 
+	// it fires whenever you slick on a DOM element with a class starting with 'build_'
+	$(".block[class*='build_']").click(function() {
+		// create an array from all of the classes in the element
+		var classes = $(this).attr("class").split(" ");
+		//loop through the array
+		classes.forEach(function(c) {
+			// check that the class starts with 'build_'. Nothing happens if no matching classes are found.
+			if (c.substring(0, 6) == "build_") {
```

Comment:
It's because I've had to use a universal selector query ("class*****='build_'")  instead of a start of line selector query ("class**^**='build_'"). 

In my testing, the start of line selector wasn't working, and a believe it's because the build_whatever classes are not the first class on the elements, ie. class="somethingelse build_whatever". the universal selector, however, worked fine. 

I don't know enough about the more complex CSS selectors to debug it so adding extra validation was an easy solution.

Also it it never hurts to over-validate and ensure you are working on the right class, especially with selectors as broad as these.